tools/snap/build_remote.py

```diff
@@ -0,0 +1,151 @@
+    while True:
+        if final:
+            print('Results for remote build finished at {0}'.format(datetime.datetime.now()))
+        else:
+            print('Remote build status at {0}'.format(datetime.datetime.now()))
+        print(' project                    {0}'.format(''.join('| {0}                       '.format(arch)
+                                                               for arch in archs)))
+        print('----------------------------{0}'.format('+-----------------------------' * len(archs)))
+        for project, states in sorted(status.items()):
+            print(' {0} {1}'.format(
+                project + ' ' * (25 - len(project)),
+                ''.join(' | {0}'.format(states[arch] + ' ' * (27 - len(states[arch]))) for arch in archs)))
+        print('----------------------------{0}'.format('+-----------------------------' * len(archs)))
+        print()
+
+        sys.stdout.flush()
+
+        if final:
+            break
+
+        time.sleep(10)
+
+
+def _dump_results(targets, archs, status, workspaces):
+    failures = False
+    for target in targets:
+        for arch in archs:
+            result = status[target][arch]
+
+            if result != 'Successfully built':
+                failures = True
+
+                with open(join(workspaces[target], '{0}_{1}.txt'.format(target, arch))) as file_h:
+                    build_output = file_h.read()
+
+                print('Output for failed build target={0} arch={1}'.format(target, arch))
+                print('-------------------------------------------')
+                print(build_output)
+                print('-------------------------------------------')
+                print()
+
+    if not failures:
+        print('All builds succeeded.')
+    else:
+        print('Some builds failed.')
+
+    return failures
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('targets', nargs='+', choices=['ALL', 'DNS_PLUGINS', 'certbot', *PLUGINS],
+                        help='the list of snaps to build')
+    parser.add_argument('--archs', nargs='+', choices=['amd64', 'arm64', 'armhf'], default='amd64',
+                        help='the architectures for which snaps are built')
+    args = parser.parse_args()
+
+    archs = set(args.archs)
+    targets = set(args.targets)
+
+    if 'ALL' in targets:
+        targets.remove('ALL')
+        targets.update(['certbot', 'DNS_PLUGINS'])
+
+    if 'DNS_PLUGINS' in targets:
+        targets.remove('DNS_PLUGINS')
+        targets.update(PLUGINS)
+
+    status = Manager().dict()
+
+    state_process = Process(target=_dump_status, args=(archs, status,))
+    state_process.start()
+
+    pool = Pool(processes=len(targets))
+    async_results = [pool.apply_async(_build_snap, (target, archs, status)) for target in targets]
+
+    workspaces = {}
+    for async_result in async_results:
+        workspaces.update(async_result.get())
+
+    state_process.terminate()
```

Comment:
I think on Linux things are working nicely because it is a `SIGTERM` that is sent. It would not be on Windows.

However I find your approach more elegant, so I implemented it. I also split the function into to dedicated `_dump_status` and `_dump_status_helper`.